File: 0145-binary-tree-postorder-traversal/0145-binary-tree-postorder-traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
void PreOrder(TreeNode* temp,vector<int>&ans){
  if(temp == NULL)return;

  PreOrder(temp->left,ans);
   PreOrder(temp->right,ans);
   ans.push_back(temp->val);
}
    vector<int> postorderTraversal(TreeNode* root) {
        vector<int>ans;
        PreOrder(root,ans);
        return ans;
    }
};

File: 0145-binary-tree-postorder-traversal/0145-binary-tree-postorder-traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {

    void help(List<Integer> ans,TreeNode root ){
        if(root == null) return;

        help(ans,root.left);
        help(ans,root.right);
        ans.add(root.val);
        return;
    }
    public List<Integer> postorderTraversal(TreeNode root) {
        List<Integer> ans = new ArrayList<>();
        help(ans,root);
        return ans;
    }
}